### ISSUE TITLE
POC Nuxt 3

### ISSUE THREAD

**Issue by ctwhome** [research-software-directory/RSD-as-a-service#29]
Investigate the possibility to use Vue 3 / Nuxt 3 at this moment.

## Motivation

Vue 3 is released about 1 year ago. Last version, at the moment of writing, is 3.2. The Vue ecosystem is slowly moving to  version 3. Nuxt is one of big players in the Vue ecosystem. RSD project aims at maximal SEO support and therefore has need for SSR features. Nuxt is the most viable option in Vue world if one does not desire to develop it's own SSR 'framework'.

## Conclusion

After my initial research about the current state of Vue "ecosystem", we decided to explore the state of Nuxt 3 beta version in order to estimate if new version can be used for production from January 2022. This could enable us to release RSD product in Q2 2022, running new Nuxt version, which at that moment might  also enter the official release state.

Nuxt 3 beta works reasonably well. However, most of the modules that make development in Nuxt very easy are not yet in beta stage. This means that most of the functionality would need to be developed using generic libraries. At the moment there is no UI framework that can be used with Nuxt 3.

To sum up, although Nuxt 3 beta can be used the rest of Nuxt ecosystem is not yet on new version which requires lot of custom development.

More information about my experience with Nuxt 3 is avaliable in README.md file in [POC Nuxt 3 branch](https://github.com/research-software-directory/RSD-as-a-service/tree/POC-nuxt-v3/frontend)
